packages/web-react/src/components/Grid/useGridItemStyleProps.ts

```diff
@@ -0,0 +1,64 @@
+import { ElementType, CSSProperties } from 'react';
+import { useClassNamePrefix } from '../../hooks';
+import { GridItemPosition, GridItemPositionBreakpoints, SpiritGridItemProps } from '../../types';
+
+interface GridItemCSSProperties extends CSSProperties {
+  [key: string]: string | undefined | number;
+}
+
+const setStyleProperty = (styleObject: GridItemCSSProperties, propertyName: string, value: string | undefined) => {
+  (styleObject as Record<string, string | undefined>)[propertyName] = value;
+};
+
+const setGridItemStyle = (styleObject: GridItemCSSProperties, baseVarName: string, propValue: GridItemPosition) => {
+  if (typeof propValue === 'object' && propValue !== null) {
+    Object.keys(propValue).forEach((key) => {
+      const suffix = key === 'mobile' ? '' : `-${key}`;
+      const propName = `--${baseVarName}${suffix}`;
+      setStyleProperty(styleObject, propName, propValue[key as keyof GridItemPositionBreakpoints]?.toString());
+    });
+  } else {
+    const propName = `--${baseVarName}`;
+    setStyleProperty(styleObject, propName, propValue.toString());
+  }
+};
+
+export interface GridItemStyles<T> {
```

Comment:
Should this be in a `types`?

Reply:
In all use hooks, we define it directly. We can move this, but we should do it globally.